server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -202,36 +229,121 @@ export class TenantManager {
     /**
      * Retrieves the secret for the given tenant
      */
-    public async getTenantKey(tenantId: string, includeDisabledTenant = false): Promise<string> {
-        const encryptedTenantKey = (await this.getTenantDocument(tenantId, includeDisabledTenant)).key;
-        const tenantKey = this.secretManager.decryptSecret(encryptedTenantKey);
-        if (tenantKey == null) {
-            winston.error("Tenant key decryption failed.");
-            Lumberjack.error("Tenant key decryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
-            return Promise.reject(new Error("Tenant key decryption failed."));
+    public async getTenantKeys(tenantId: string, includeDisabledTenant = false): Promise<ITenantKeys> {
+        const tenantDocument = await this.getTenantDocument(tenantId, includeDisabledTenant);
+        
+        const encryptedTenantKey1 = tenantDocument.key;
+        const tenantKey1 = this.secretManager.decryptSecret(encryptedTenantKey1);
+        if (tenantKey1 == null) {
+            winston.error("Tenant key1 decryption failed.");
+            Lumberjack.error("Tenant key1 decryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
+            return Promise.reject(new Error("Tenant key1 decryption failed."));
```

Comment:
Here again, if the decryption of one the two keys fails, won't it be okay?

Reply:
As mentioned with Henrique in this PR, if a user calls getKeys API, we should return both valid key1 and key2. If either of them fails, we should reject and the user would send the request again.

Reply:
I see..got it..just saw the other comments on this PR..thanks for clarifying again.